modules/govuk_jenkins/manifests/jobs/athena_fastly_logs_check/icinga_database_check.pp

```diff
@@ -0,0 +1,36 @@
+# == Define: govuk_jenkins::jobs::athena_fastly_logs_check::icinga_database_check
+#
+# Definition used to iterate through an array of database names and set-up
+# Icinga passive checks for each one of these.
+#
+# This should only have to exist as a seperate defined resource until the
+# eventual day that we upgrade Puppet to allow the future syntax where we
+# can iterate through the array.
+#
+# === Parameters
+#
+# [*jenkins_url*]
+#
+# A URL to the Jenkins job which is linked to by this alert. This URL will
+# be given a query string to filter the jobs listed to only those for this
+# particular database.
+#
+# [*service_description*]
+#
+# The description that is shown in Icinga for this check. It is expected to
+# have a variable interpolated of $DATABASE, this is replaced for the database
+# name that is under check.
+define govuk_jenkins::jobs::athena_fastly_logs_check::icinga_database_check (
+  $jenkins_url = undef,
+  $service_description = undef,
+) {
+  @@icinga::passive_check {
+    "athena_fastly_logs_check_${name}_${::hostname}":
```

Comment:
I'm not quite sure where `${name}` is coming from here.

Reply:
Thats the variable that is automatically set with each item in the array that's passed in. For example: https://stackoverflow.com/a/13008766

This define was a total pain to deal with due to puppet 3's lack of array iteration abilities. Do you know of any examples in our code base that are clearer to understand?

Reply:
Ah, OK, I'm used to the magical `title` variable, but obviously I've overlooked the `name` one :+1:

Reply:
So `${name}` is `govuk_www` or `govuk_assets`?

Reply:
Yup - that's correct Leena.